ARCH.md

```diff
@@ -0,0 +1,27 @@
+The LDK sample is broken between 2 primary process : `ldk-node` and `ldk-cli`.
+
+The `ldk-cli` is a simple JSON-RPC client to interact with the node.
+
+The `ldk-node` is a multi-threaded process embedding all the main components of a regular Lightning
+node :
+* `MicroSPVClient`, a chain validation backend providing blocks updates, utxo filtering and utxo set access
+* `ChainMonitor`, a chain processing entity enforcing onchain the channel logic negotiated offchain
+* `EventHandler`, a default handler informing the client logic about internal events
+* `ChannelManager`, a core lightning state machine, supervising channel lifecycle and payment initiation/relay
+* `NetGraphMsgHandler`, a lightning router, handling gossips updates and providing payment path draws
+* `PeerManager`, a lightning networking stack, managing peer connections lifecycle
+* `setup_socket_listener`, a simple network handler to listen incoming connections
+* `setup_rpc_server`, a simple RPC server handler to listen to `ldk-cli`
+
+All those components are running in their own threads and communicate between them using inter-threading
+handlers powered by `tokio::sync::mpsc`. As the communication channels provided by tokio are unidirectional,
+if the aforementioned components require bidirectional flows (e.g `ChannelManager` asking a route to
+`NetGraphMsgHandler`) a pair of inter-threading handlers are provided.
+
+# Why the sharing messages approach over the sharing memory one ?
+
+Long-term, it might be interesting to provide a multi-process LDK sample, better fitted for
```

Comment:
Note that a big part of the motivation for this in Core is that its written in a memory-unsafe language. Without unsafe code, there's little reason to do this (absent compiler and other similar bugs). Cross-host, not just cross-process, stuff is cooler, though.

Reply:
Come on, playing with the security capabilities or privileges framework of your system is cool too. Or even control groups for intensive process like routing or throwing a load balancer between p2p stack and channel processing. For high-availability, high-security node, multi-process is way better :p

Reply:
Lol, I mean I guess if you want capabilities enforcement that's cool. I still prefer multi-host, but whatever :)